### PR TITLE
Add A SLD local loader.

### PR DESCRIPTION
**Pull request by RemiDesgrange**
Following #78  It still need a PHP backend to come.

### CODE REVIEW

File: js/mviewerstudio.js [hunk 1]
```diff
@@ -909,3 +909,20 @@ $(".checkedurl").change(mv.checkURL);
...
+    var sldFile = e.files[0];
+    $.ajax('/srv/store/style', {
+        data: sldFile,
+        method: 'POST',
+        processData: false,
+        success: function(data) {
+            $("#frm-sld").val(data.filepath)
+        },
+        error: function() {
+            alert("Le fichier n'a pas pu être chargé")
```
Comment (Gaetanbrl) [inline js/mviewerstudio.js:924]: Needs i18n like this ? 
https://github.com/geobretagne/mviewerstudio/blob/052a667decb8f714ea62a90ac2a21ace58b1d043/js/mviewerstudio.js#L773

File: index.html [hunk 2]
```diff
@@ -687,9 +687,28 @@ <h4 class="modal-title" i18n="modal.layer.edit">Edition de couche</h4>
...
+                                                        i18n="modal.layer.uploadsldfile"
+                                                        onclick="document.getElementById('frm-upload-sld-file').click()"
+                                                >
+                                                    Téléverser un SLD
+                                                </button>
+                                                <button class="btn btn-default" type="button">
+                                                    <span class="glyphicon glyphicon-info-sign"
+                                                          data-toggle="tooltip"
+                                                          data-placement="top"
+                                                          title="Vous pouvez insérer une URL vers un SLD. Si votre couche provient d'un geoserver, vous pouvez specifier uniquement le nom du SLD dans geoserver (ex: un_entrepot:mon_sld)"
```
Comment (Gaetanbrl) [inline index.html:706]: same comment for i18n ?

Comment (RemiDesgrange) [reply index.html:706]: I have no idea how to make this work with i18njs though

Comment (Gaetanbrl) [reply index.html:706]: 10 lines above ;)

https://github.com/geobretagne/mviewerstudio/blob/aaaae8e45ad927f8e509c87fda319c3fb32f654a/index.html#L697

**Comment by RemiDesgrange** [comment]
Test this PR is not that easy. Since the SLD is passed to a geoserver instance, the geoserver where you pass the URL of SLD need to be able to see the mviewer instance. This particularity is easy to meet in production but quite hard in a dockerized env in dev. You'll need to have a local geoserver running in order to test that. A proxy is also required.

**Comment by RemiDesgrange** [comment]
I resolved the conflict locally and pushed to camptocamp/mviewerstudio, but the conflict still appear, is it a glitch or something ?

**Comment by fvanderbiest** [comment]
What's missing here ? Can it be merged ?

**Pull request merged**